client/cypress/end-to-end/ballot-comparison-audit-board.spec.js

```diff
@@ -0,0 +1,287 @@
+    cy.wait(1000)
+    cy.logout()
+    cy.contains('Participating in an audit in your local jurisdiction?')
+    cy.loginJurisdictionAdmin('wtarkin@empire.gov')
+    cy.findByText(`Jurisdictions - TestAudit${id}`).siblings('button').click()
+    cy.fixture('CSVs/manifest/ballot_comparison_manifest.csv').then(fileContent => {
+      cy.get('input[type="file"]').first().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'ballot_comparison_manifest.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').spread((firstButton, secondButton) => {
+      firstButton.click()
+    })
+    cy.contains("Upload successfully completed")   
+
+    cy.fixture('CSVs/cvr/ballot_comparison_cvr.csv').then(fileContent => {
+      cy.get('input[type="file"]').last().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'ballot_comparison_cvr.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').click()
+    cy.findAllByText(/Upload successfully completed/).should('have.length', 2)
+    cy.wait(1000)
+    cy.logout()
+    cy.loginAuditAdmin('audit-admin-cypress@example.com')
+    cy.findByText(`TestAudit${id}`).click()
+    cy.findByText('Review & Launch').click()
+    cy.findByText('Launch Audit').click()
+    cy.findAllByText('Launch Audit').spread((firstButton, secondButton) => {
+      secondButton.click()
+    })
+    cy.logout()
+    cy.wait(1000)
+    cy.contains('Participating in an audit in your local jurisdiction?')
+
+  })
+  const board_credentials_url = ''
+
+  it('Audit Board Completion', () => {
+    cy.loginJurisdictionAdmin('wtarkin@empire.gov')
+    cy.findByText(`Jurisdictions - TestAudit${id}`).siblings('button').click()
+    cy.contains('Number of Audit Boards')
+    cy.findByText('Save & Next').click()
+    cy.findByText('Download Audit Board Credentials').click()
+    cy.logout()
+    cy.wait(1000)
+    cy.task('getPdfContent', `cypress/fixtures/PDFs/Audit Board Credentials\ -\ Death Star\ -\ TestAudit${id}.pdf`).then((content) => {
+      function urlify(text) {
+        var urlRegex = /(((https?:\/\/)|(www\.))[^\s]+)/g;
+        return text.match(urlRegex, function(url) {
+          return url
+        }) 
+      }
+      board_credentials_url = urlify(content.text);
+      cy.visit(board_credentials_url[0])
+      cy.findAllByText('Audit Board Member').eq(0).siblings('input').type('Board Member 1')
+      cy.findAllByText('Audit Board Member').eq(1).siblings('input').type('Board Member 2')
+      cy.findByText('Next').click()
+      cy.contains(/Ballot Cards to Audit/)
+      cy.get('table tbody tr').each(($el, index, list) => {
+          if(index == 0) {
+              cy.findByText('Start Auditing').click()
+          }
+          cy.get('input[type="checkbox"]').first().click({force: true})
+          cy.findByText('Review').click() 
+          cy.findByText('Submit & Next Ballot').click() 
+      })
```

Comment:
This particular bit confused me - it took a while for me to figure out why you were looping over the rows of the table, since you're actually not on that screen anymore once you click "Start Auditing".

As a user, I think I would just click "Start Auditing" and then audit all of the ballots until it took me back to the main screen with the table of ballots. A way to implement that would be to define a function `auditBallot` that audits one currently shown ballot, clicks "Submit & Next Ballot", and then checks to see whether it's looking at another ballot or if it's back on the main screen. If it's looking at another ballot, it can call `auditBallot` again recursively. Otherwise it can return and the test script can continue. (I based this idea on this: https://github.com/cypress-io/cypress/issues/8100#issuecomment-664919436)

Or if you want to stick with this, maybe you could add some comments explaining what's going on

Reply:
Yeah, I see what you're saying. I'll see if I can find a more straightforward and simple way to do it, and refactor to that or add comments.